blueflood-core/src/main/java/com/rackspacecloud/blueflood/cache/TokenCache.java

```diff
@@ -0,0 +1,75 @@
+package com.rackspacecloud.blueflood.cache;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.rackspacecloud.blueflood.types.Token;
+import com.rackspacecloud.blueflood.utils.Metrics;
+
+import java.util.concurrent.TimeUnit;
+
+public class TokenCache {
+
+    // this collection is used to reduce the number of tokens that get written.
+    // Simply, if a token has been seen within the last 10 minutes, don't bother.
```

Comment:
In `TokenCache`, I dont cache leaf node as a leaf node is same a locator name. I use `LocatorCache` for leaf nodes and for non-leaf nodes use `TokenCache`. This logic is in `TokenDiscoveryWriter.processTokens()` method.